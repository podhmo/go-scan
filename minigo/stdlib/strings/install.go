// Code generated by minigo-gen-bindings. DO NOT EDIT.

package strings

import (
	"strings"
	"github.com/podhmo/go-scan/minigo"
)

// Install binds all exported symbols from the "strings" package to the interpreter.
func Install(interp *minigo.Interpreter) {
	interp.Register("strings", map[string]any{
		"Clone": strings.Clone,
		"Compare": strings.Compare,
		"Contains": strings.Contains,
		"ContainsAny": strings.ContainsAny,
		"ContainsFunc": strings.ContainsFunc,
		"ContainsRune": strings.ContainsRune,
		"Count": strings.Count,
		"Cut": strings.Cut,
		"CutPrefix": strings.CutPrefix,
		"CutSuffix": strings.CutSuffix,
		"EqualFold": strings.EqualFold,
		"Fields": strings.Fields,
		"FieldsFunc": strings.FieldsFunc,
		"FieldsFuncSeq": strings.FieldsFuncSeq,
		"FieldsSeq": strings.FieldsSeq,
		"HasPrefix": strings.HasPrefix,
		"HasSuffix": strings.HasSuffix,
		"Index": strings.Index,
		"IndexAny": strings.IndexAny,
		"IndexByte": strings.IndexByte,
		"IndexFunc": strings.IndexFunc,
		"IndexRune": strings.IndexRune,
		"Join": strings.Join,
		"LastIndex": strings.LastIndex,
		"LastIndexAny": strings.LastIndexAny,
		"LastIndexByte": strings.LastIndexByte,
		"LastIndexFunc": strings.LastIndexFunc,
		"Lines": strings.Lines,
		"Map": strings.Map,
		"NewReader": strings.NewReader,
		"NewReplacer": strings.NewReplacer,
		"Repeat": strings.Repeat,
		"Replace": strings.Replace,
		"ReplaceAll": strings.ReplaceAll,
		"Split": strings.Split,
		"SplitAfter": strings.SplitAfter,
		"SplitAfterN": strings.SplitAfterN,
		"SplitAfterSeq": strings.SplitAfterSeq,
		"SplitN": strings.SplitN,
		"SplitSeq": strings.SplitSeq,
		"Title": strings.Title,
		"ToLower": strings.ToLower,
		"ToLowerSpecial": strings.ToLowerSpecial,
		"ToTitle": strings.ToTitle,
		"ToTitleSpecial": strings.ToTitleSpecial,
		"ToUpper": strings.ToUpper,
		"ToUpperSpecial": strings.ToUpperSpecial,
		"ToValidUTF8": strings.ToValidUTF8,
		"Trim": strings.Trim,
		"TrimFunc": strings.TrimFunc,
		"TrimLeft": strings.TrimLeft,
		"TrimLeftFunc": strings.TrimLeftFunc,
		"TrimPrefix": strings.TrimPrefix,
		"TrimRight": strings.TrimRight,
		"TrimRightFunc": strings.TrimRightFunc,
		"TrimSpace": strings.TrimSpace,
		"TrimSuffix": strings.TrimSuffix,
	})
}
