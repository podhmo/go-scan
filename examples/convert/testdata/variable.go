package testdata

import "strings"

// // convert:variable builder strings.Builder
// @derivingconvert(DstVariable)
type SrcVariable struct {
	FirstName string
	LastName  string
}

type DstVariable struct {
	FullName string `convert:",using=buildFullName(&builder, src.FirstName, src.LastName)"`
}

func buildFullName(builder *strings.Builder, firstName, lastName string) string {
	builder.Reset()
	builder.WriteString(firstName)
	builder.WriteString(" ")
	builder.WriteString(lastName)
	return builder.String()
}
