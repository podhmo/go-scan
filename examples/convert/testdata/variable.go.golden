// Code generated by convert. DO NOT EDIT.
package testdata

import (
	"context"
	"errors"
	"strings"

	"github.com/podhmo/go-scan/examples/convert/model"
)

func convertSrcVariableToDstVariable(ctx context.Context, ec *model.ErrorCollector, src *SrcVariable) *DstVariable {
	if src == nil {
		return nil
	}
	var builder strings.Builder
	dst := &DstVariable{}
	return dst
}

// ConvertSrcVariableToDstVariable converts SrcVariable to DstVariable.
//
// Fields that are not populated by this converter:
//   - FullName
func ConvertSrcVariableToDstVariable(ctx context.Context, src *SrcVariable) (*DstVariable, error) {
	if src == nil {
		return nil, nil
	}
	ec := model.NewErrorCollector(0)
	dst := convertSrcVariableToDstVariable(ctx, ec, src)
	if ec.HasErrors() {
		return dst, errors.Join(ec.Errors()...)
	}
	return dst, nil
}
